spring-cloud-kubernetes-client-discovery/src/main/java/org/springframework/cloud/kubernetes/client/discovery/reactive/KubernetesInformerReactiveDiscoveryClientAutoConfiguration.java

```diff
@@ -43,39 +40,30 @@
 import org.springframework.cloud.client.discovery.health.reactive.ReactiveDiscoveryClientHealthIndicator;
 import org.springframework.cloud.client.discovery.simple.reactive.SimpleReactiveDiscoveryClientAutoConfiguration;
 import org.springframework.cloud.kubernetes.client.KubernetesClientPodUtils;
-import org.springframework.cloud.kubernetes.client.discovery.KubernetesDiscoveryClientAutoConfiguration;
+import org.springframework.cloud.kubernetes.client.discovery.CatalogSharedInformerFactory;
+import org.springframework.cloud.kubernetes.client.discovery.SpringCloudKubernetesInformerFactoryProcessor;
 import org.springframework.cloud.kubernetes.commons.KubernetesNamespaceProvider;
 import org.springframework.cloud.kubernetes.commons.discovery.ConditionalOnKubernetesDiscoveryEnabled;
 import org.springframework.cloud.kubernetes.commons.discovery.KubernetesDiscoveryProperties;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
+import org.springframework.core.env.Environment;
 
 /**
  * @author Ryan Baxter
  */
 
 @Configuration(proxyBeanMethods = false)
```

Comment:
as we the blocking implementation, I wanted a few clear things here:

- proper annotations:

```
@ConditionalOnDiscoveryEnabled
@ConditionalOnKubernetesDiscoveryEnabled
@ConditionalOnReactiveDiscoveryEnabled
```

- do not depend on any beans coming from blocking implementation (as such remove `KubernetesDiscoveryClientAutoConfiguration.class` from `@AutoConfigureAfter`). The reason is that someone can disable blocking and then reacive auto-configuration will fail. 
- add `@EnableConfigurationProperties(KubernetesDiscoveryProperties.class)`. This one was again an artifact of the previous implementation, as it was coming from the blocking auto-configuration. 
-